neqo-http3/src/connection.rs

```diff
@@ -1304,7 +1298,7 @@ mod tests {
 
         // assert no error occured.
         assert_eq!(hconn.state(), Http3State::Connected);
-        (hconn, neqo_trans_conn, control_stream)
+        (hconn, neqo_trans_conn, control_stream, encoder)
```

Comment:
This function is now a little hard to read.  Could we perhaps make a test peer connection object that holds the neqo-transport::Connection, control stream ID, and QPackEncoder?